### PR TITLE
style-guide: update testing links

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: style-guide: update testing links
Additional Description:
minor update to some links in the style guide.

Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: STYLE.md [hunk 1]
```diff
@@ -233,8 +233,8 @@ environment. In general, there should be no non-local network access. In additio
 
 Tests should be deterministic. They should not rely on randomness or details
 such as the current time. Instead, mocks such as
-[`MockRandomGenerator`](test/mocks/runtime/mocks.h) and
-[`Mock*TimeSource`](test/mocks/common.h) should be used.
+[`MockRandomGenerator`](test/mocks/common.h) and
+[`SimulatedTimeSystem`](test/test_common/simulated_time_system.h) should be used.
```
Comment (tyxia) [inline STYLE.md:237]: nit: maybe `SimulatedTimeSystem is recommended` ?
I know real test time is still used in some places where inject-time mechanism is not provided. e.g., gRPC

**Review by tyxia — APPROVED** [review]
LGTM, modulo one comment.

Thanks!

File: STYLE.md (hunk 1 above)
Comment (adisuissa) [reply STYLE.md:237]: `should` implies it is recommended.
I'm assuming some randomness cannot be removed completely (from used libraries for example), but this should be the guideline for most tests.

Comment (tyxia) [reply STYLE.md:237]: SGTM

**Review by tyxia — APPROVED** [review]

**Pull request merged**